pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTableRestletResource.java

```diff
@@ -712,6 +719,37 @@ public String getTableState(
     }
   }
 
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  @Authenticate(AccessType.UPDATE)
+  @Path("/rebalanceStatus/{jobId}")
+  @ApiOperation(value = "Gets detailed stats of a rebalance operation",
+          notes = "Gets detailed stats of a rebalance operation")
+  public ServerRebalanceJobStatusResponse rebalanceStatus(@ApiParam(value = "Rebalance Job Id", required = true)
+                                         @PathParam("jobId") String jobId)
+          throws JsonProcessingException {
+    Map<String, String> controllerJobZKMetadata =
+            _pinotHelixResourceManager.getControllerJobZKMetadataResource(jobId,
+                    CommonConstants.Helix.ZkClient.ZK_REBALANCE_RESOURCE);
+    if (controllerJobZKMetadata == null) {
+      throw new ControllerApplicationException(LOGGER, "Failed to find controller job id: " + jobId,
+              Response.Status.NOT_FOUND);
+    }
+    TableRebalanceProgressStats tableRebalanceProgressStats = JsonUtils.stringToObject(
+            controllerJobZKMetadata.get(RebalanceConfigConstants.REBALANCE_PROGRESS_STATS),
+            TableRebalanceProgressStats.class);
+    Long timeSinceStartInSecs = 0L;
+    if (!tableRebalanceProgressStats.getStatus().toString().equals(RebalanceResult.Status.DONE.toString())) {
+      timeSinceStartInSecs = (long) (System.currentTimeMillis()
+              - tableRebalanceProgressStats.getStartTimeInMilliseconds()) / 1000;
+    }
+    ServerRebalanceJobStatusResponse serverRebalanceJobStatusResponse = new ServerRebalanceJobStatusResponse();
+    serverRebalanceJobStatusResponse.setTableRebalanceProgressStats(tableRebalanceProgressStats);
+    serverRebalanceJobStatusResponse.setTimeElapsedSinceStartInSeconds(timeSinceStartInSecs);
+    return serverRebalanceJobStatusResponse;
```

Comment:
Does this include a response code (eg 200 ok)?

Reply:
Yes response code is automatically included.  Snippet of the output

Code | Details
-- | --
200.   Response body
Download
{
  "tableRebalanceProgressStats": {
    "initialToTargetStateConvergence": {
      "_segmentsMissing": 0,
      "_segmentsToRebalance": 31,
      "_percentSegmentsToRebalance": 100,
      "_replicasToRebalance": 279
    },
    "externalViewToIdealStateConvergence": {
      "_segmentsMissing": 0,
      "_segmentsToRebalance": 31,
      "_percentSegmentsToRebalance": 100,
      "_replicasToRebalance": 279
    },
    "currentToTargetConverg